src/utilities/vizSupport.py

```diff
@@ -1037,6 +1039,15 @@ def createCameraConfigMsg(viz, **kwargs):
     else:
         cameraConfigMsgPayload.cameraID = 0
 
+    if 'renderMode' in kwargs:
+        val = kwargs['renderMode']
+        if not isinstance(val, int) or val < 0:
+            print('ERROR: vizSupport: renderMode must be 0 or 1.')
+            exit(1)
+        cameraConfigMsgPayload.renderMode = val
```

Comment:
:large_orange_diamond: Given the description in the comment above, it would be semantically clearer for users and less error prone to have the variable named `renderDepthBuffer` or something similar and the value be a bool. 

Similarly, in the above value validation we are checking if the val is an `'int` and positive, yet that still allows the value to be 2,3,... We would provide a better experience if we didn't silently swallow invalid values. 

Something I learnt a few years ago which I find extremely valuable is the idea "may forbidden states unrepresentable" . That is, don't allow users to express something that is not defined/allowed/part of the semantics.

Reply:
I'll update the checking to be within the valid range of numbers.

Reply:
Are you proposing to rename `renderMode` to `renderDepthBuffer`?  We chose to use `renderMode` as the camera will either provide a visible light image, or the depth map.   Calling it `renderMode` kept it general in case in the future we have other camera render modes (IR, false color, etc.). Right now there are only 2 modes, thus the options are 0 and 1.

Reply:
Yes that was my proposition. But leaving it as `renderMode` is okay too. But I guess in that case it doesn't need to be a render one frame buffer or another )I'm not proposing to change the already developed capability, rather to think about it's implementation choices down the track). Rather, if we are hoping to return a variety of framebuffer contents then the Viz could be instructed to render each of the requested frame. This is what OpenGL/DirectX is doing underneath. So in theory, the viz could be told to produce a number of different renders, visible, depth, etc. and return/make available each of these for each time step. So it would be less of a mode, and more of a "hey, render/return each of these framebuffers".